apps/passport-client/pages/index.tsx

```diff
@@ -71,251 +68,226 @@ import { AppState, StateEmitter } from "../src/state";
+        const checkinOfflineTicketsResult = await requestOfflineTicketsCheckin(
+          appConfig.zupassServer,
+          {
+            checkedOfflineInDevconnectTicketIDs:
+              state.checkedinOfflineDevconnectTickets.map((t) => t.id),
+            checkerProof: await getOrGenerateCheckinCredential(state.identity)
+          }
+        );
+
+        if (checkinOfflineTicketsResult.success) {
+          update({
+            checkedinOfflineDevconnectTickets: []
+          });
+          saveCheckedInOfflineTickets(undefined);
+        }
+      }
+
+      const offlineTicketsResult = await requestOfflineTickets(
         appConfig.zupassServer,
         {
-          checkedOfflineInDevconnectTicketIDs:
-            this.state.checkedinOfflineDevconnectTickets.map((t) => t.id),
-          checkerProof: await getOrGenerateCheckinCredential(
-            this.state.identity
-          )
+          checkerProof: await getOrGenerateCheckinCredential(state.identity)
         }
       );
 
-      if (checkinOfflineTicketsResult.success) {
-        this.update({
-          ...this.state,
-          checkedinOfflineDevconnectTickets: []
+      if (offlineTicketsResult.success) {
+        update({
+          offlineTickets: offlineTicketsResult.value.offlineTickets
         });
-        saveCheckedInOfflineTickets(undefined);
-      }
-    }
-
-    const offlineTicketsResult = await requestOfflineTickets(
-      appConfig.zupassServer,
-      {
-        checkerProof: await getOrGenerateCheckinCredential(this.state.identity)
+        saveOfflineTickets(offlineTicketsResult.value.offlineTickets);
       }
-    );
+    };
 
-    if (offlineTicketsResult.success) {
-      this.update({
-        ...this.state,
-        offlineTickets: offlineTicketsResult.value.offlineTickets
+    const startBackgroundJobs = (): void => {
+      console.log("[JOB] Starting background jobs...");
+      document.addEventListener("visibilitychange", () => {
+        setupPolling();
       });
-      saveOfflineTickets(offlineTicketsResult.value.offlineTickets);
-    }
-  };
+      setupPolling();
+      startJobSyncOfflineCheckins();
+      jobCheckConnectivity();
+      generateCheckinCredential();
+    };
+
+    setupBroadcastChannel(dispatch);
+    setupUsingLaserScanning();
+    startBackgroundJobs();
+
+    return () => {
+      closeBroadcastChannel();
+    };
+  });
+}
+
+function App(): JSX.Element {
+  useBackgroundJobs();
+  const state = useStateContext().getState();
+
+  const hasStack = !!state.error?.stack;
+  return (
+    <>
+      {!isWebAssemblySupported() ? (
+        <HashRouter>
+          <Routes>
+            <Route path="/terms" element={<TermsScreen />} />
+            <Route path="*" element={<NoWASMScreen />} />
+          </Routes>
+        </HashRouter>
+      ) : !hasStack ? (
+        <Router />
+      ) : (
+        <HashRouter>
+          <Routes>
+            <Route path="*" element={<AppContainer bg="gray" />} />
```

Comment:
this may or may not be an inherited mistake but i wonder if the `/terms` page should be included in the routes here?